src/components/FeedbackWidget.js

```diff
@@ -0,0 +1,254 @@
+    transition: transform 0.2s ease-in-out;
+  }
+  transition: transform 0.2s ease-in-out;
+
+  flex-direction: column;
+  align-items: center;
+  text-align: center;
+  padding: 2rem;
+
+  .title {
+    font-weight: 700;
+    font-size: 1.25rem;
+    line-height: 1.5rem;
+  }
+
+  .subtitle {
+    font-weight: 400;
+    font-size: 1rem;
+    line-height: 1.25rem;
+  }
+
+  .timing {
+    font-weight: 700;
+    font-size: 0.75rem;
+    line-height: 1rem;
+    letter-spacing: 0.025em;
+    color: ${({ theme }) => theme.colors.ghostCardGhost};
+  }
+
+  a {
+    width: 100%;
+    color: ${({ theme }) => theme.colors.text} !important;
+  }
+`
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  gap: 1.25rem;
+  width: 100%;
+  * {
+    flex: 1;
+    color: ${({ theme }) => theme.colors.white};
+    font-weight: 700;
+  }
+`
+
+const StyledFeedbackGlyph = styled(FeedbackGlyph)`
+  path {
+    fill: ${({ theme }) => theme.colors.white};
+  }
+`
+
+const IconContainer = styled.div`
+  position: absolute;
+  right: 0.5rem;
+  top: 0.5rem;
+  cursor: pointer;
+  &:hover {
+    transform: scale(1.1);
+    transition: transform 0.2s ease-in-out;
+  }
+  transition: transform 0.2s ease-in-out;
+`
+
+const FeedbackWidget = ({ prompt, className }) => {
+  const containerRef = useRef()
+  useOnClickOutside(containerRef, () => handleClose(), [`mousedown`])
+  const [isOpen, setIsOpen] = useState(false)
+  const [feedbackSubmitted, setFeedbackSubmitted] = useState(false)
+  const [isHelpful, setIsHelpful] = useState(null)
+
+  const handleClose = () => {
+    setIsOpen(false)
+    trackCustomEvent({
+      eventCategory: `FeedbackWidget toggled`,
+      eventAction: `Clicked`,
+      label: `Closed feedback widget`,
+    })
+  }
+  const handleOpen = () => {
+    setIsOpen(true)
+    trackCustomEvent({
+      eventCategory: `FeedbackWidget toggled`,
+      eventAction: `Clicked`,
+      label: `Opened feedback widget`,
+    })
+  }
+  const handleSubmit = (choice) => {
+    trackCustomEvent({
+      eventCategory: `Page is helpful feedback`,
+      eventAction: `Clicked`,
+      label: choice,
+    })
+    setIsHelpful(choice)
+    setFeedbackSubmitted(true)
+  }
+
+  const location = typeof window !== "undefined" ? window.location.href : ""
+  const isStaking = location.includes("staking")
+
+  const surveyUrls = {
+    default: {
+      yes: `https://czvgzauj.paperform.co/?url=${location}`,
+      no: `https://xlljh5l3.paperform.co/?url=${location}`,
+    },
+    staking: {
+      yes: `https://gzmn3wgk.paperform.co/?url=${location}`,
+      no: `https://zlj83p6l.paperform.co/?url=${location}`,
+    },
+  }
+
+  const getSurveyUrl = () => {
+    if (!feedbackSubmitted) return null
+    if (isStaking) return surveyUrls.staking[isHelpful ? "yes" : "no"]
+    return surveyUrls.default[isHelpful ? "yes" : "no"]
+  }
+
+  const pathsWithBottomNav = ["/staking", "/dao", "/defi", "/nft"]
+
+  const bottomOffset = () => {
```

Comment:
Consider using https://reactjs.org/docs/hooks-reference.html#usememo in this case. You will receive the offset value instead of a function and you will only be calling it when `location` changes.

Reply:
Yeah good call.. I tried this originally and it caused an error but I think I figured out what I did wrong. Will patch this and push 👍🏻

Reply:
@pettinarip Hm actually, getting some issues depending on node version. With Node 18 I don't have issues, but with anything lower I get an error:

```log
  Error: Unable to deserialize cloned data due to invalid or unsupported v
  ersion.
  
  - v8.js:343 deserialize
    v8.js:343:7
  
  - serializer.js:249 deserialize
    [ethereumorg-website]/[@parcel]/core/lib/serializer.js:249:48
  
  - RequestTracker.js:854 loadRequestGraph
    [ethereumorg-website]/[@parcel]/core/lib/RequestTracker.js:854:52
  
  - RequestTracker.js:821 Function.init
    [ethereumorg-website]/[@parcel]/core/lib/RequestTracker.js:821:17
  
  - Parcel.js:256 Parcel._init
    [ethereumorg-website]/[@parcel]/core/lib/Parcel.js:256:28
  
  - Parcel.js:273 Parcel.run
    [ethereumorg-website]/[@parcel]/core/lib/Parcel.js:273:7
  
  - compile-gatsby-files.ts:48 compileGatsbyFiles
    [ethereumorg-website]/[gatsby]/src/utils/parcel/compile-gatsby-files.t
    s:48:29
  
  - initialize.ts:171 initialize
    [ethereumorg-website]/[gatsby]/src/services/initialize.ts:171:3
  

not finished compile gatsby files - 0.395s

error Command failed with exit code 1.
```

https://support.gatsbyjs.com/hc/en-us/articles/360052322954-Environment-Variables-Specific-to-Gatsby-Cloud

Looks like Gatsby defaults to 14, and I don't see an override anywhere. Going to hold on patching this for now given the node version issue. Should still work just fine without the `useMemo` just might make it a bit less efficient. Curious your take on this Pablo.

Reply:
Hum, weird error. Can't see why this is related to the node version 🤔 Will take a closer look later on this but feel free to skip this, its totally a nice to have.
Just to confirm, this is an issue you see in your local or in GC?

Reply:
Local... just committed it to give it a try. If the build fails I'll just revert

Reply:
I have a feeling the node version errors I'm getting are related to Apple silicon

Reply:
Yep, pretty sure that's it. All good here

Reply:
Cool! good to know.